docs/architecture/api-20201231-preview.md

```diff
@@ -0,0 +1,124 @@
+if these options where enabled on cluster create and set those accordingly.
+
+This validation should be part of machine-api webhooks.
+
+```
+	// The cluster security profile
+	SecurityProfile SecurityProfile `json:"securityProfile,omitempty"`
+```
+
+```
+// SecurityProfile represents an security profile
+type SecurityProfile struct {
+	// EncryptionAtHost value sets encryptionAtHost option for all VirtualMachines.
+	EncryptionAtHost *bool `json:"encryptionAtHost,omitempty"`
+
+	// ManagedDiskEncryptionSetID represents Virtual Machine managed disk DiskEncryptionSetID
+	// encryption
+	ManagedDiskEncryptionSetID string `json:"diskEncryptionSetID,omitempty"`
+}
+```
+
+### Extend OpenShiftClusterCredentials with GET
+
+Extend `OpenShiftClusterCredentials` where current behaviour (POST to get credentials), is maintained. But if GET call is executed we allow to download kubeConfig for the cluster. This would allow customer to have system credentials in a cold storage if needed and access cluster in case of oAuth or Console outage.
+
+```
+	// Downloads kubeconfig file
+	s.Methods(http.MethodGet).HandlerFunc(f.getOpenShiftClusterCredentials).Name("getOpenShiftClusterCredentials")
+
+```
+
+### Add Isolated compute SKUs
+
+For Government cloud we need to extend our current support SKUs with isolated compute SKUs. With adding of https://github.com/Azure/ARO-RP/commit/213abf54a65ff17abeebc214c862a6c6b10c6d82 to our API this should be no-op change for new APIs in the future.
+
+This has hard requirement on OpenShift being able to rotate master nodes,
+as Isolated compute SKU's are retirable, so recovery process is pre-requisite.
+
+Isolated SKUs for decision making:
+
+```
+Standard_E64is_v3
+Standard_E64i_v3
+Standard_M128ms
+Standard_GS5
+Standard_G5
+Standard_F72s_v2
+```
+# Other proposals:
+
+Other proposal is good to have, but might be out of scope for this API release.
+
+### Add maintenance profile
+
+Maintenance profile would allow to configure automated upgrade windows for the
+customer.
+
+```
+// MaintenanceProfile allows customer to configure maintenance window for automated
+// cluster upgrades
+type MaintenanceProfile struct {
+	MaintenanceWindowStartDate        string `json:"maintenanceWindowstartDate,omitempty"`
+	MaintenanceWindowStopDate         string `json:"maintenanceWindowstopDate,omitempty"`
+	MaintenanceWindowScheduleTimezone string `json:"maintenanceWindowscheduleTimezone,omitempty"`
+    // DenyAssignment is read only field representing if DenyAssignment is enabled
+    // on cluster in question
+    DenyAssignment bool `json:"denyAssignment,omitempty"`
+}
+```
+
+### Add feature flag `"Microsoft.RedHatOpenShift/DennyAssignment"`
+
+Add custom feature flag which would allow to remove `DennyAssignment` for the cluster and enable customer to experiment more with underlaying platform.
+This would be one way flag, and it would render cluster unsupported.
+
+
+
+### Networking profile
```

Comment:
I think this one is worth bottoming out to workout what we could ship and if we should ship it.

Reply:
Architecture call?

Reply:
Should we target SDNplugin for now and maybe all others later?